lib/spack/spack/binary_distribution.py

```diff
@@ -256,7 +258,7 @@ def build_tarball(spec, outdir, force=False, rel=False, yes_to_all=False,
     # create compressed tarball of the install prefix
     with closing(tarfile.open(tarfile_path, 'w:gz')) as tar:
         tar.add(name='%s' % workdir,
-                arcname='%s' % os.path.basename(workdir))
+                arcname='.')
```

Comment:
I think this was originally intended to use binary cached packages on Spack installations with different directory layouts. Is this useful for other purposes as well? I'm checking because IIRC you mentioned additional work needed to be done to support that.

Reply:
No. To install in a different directory structure requires redoing the rpaths to the dependencies for the new directory structure. I don't see a way to do that the moment. I would rather focus on replacing paths compiled into object files first.

Reply:
OK. If this change isn't being used for anything in this PR then I think it would be clearer to remove it (and reintroduce it in a PR which targets allowing relocation between different directory layouts).

Reply:
Reverting this requires also backing out the way the tarball is unpacked and the way the workdir is named. I would rather leave it in.

Reply:
Ah but I see that this breaks the unpacking of existing tarballs. I will revert and push the changes.